CSharpFunctionalExtensions/AsyncResultExtensionsLeftOperand.cs

```diff
@@ -78,7 +78,7 @@ public static async Task<Result<T>> Ensure<T>(this Task<Result<T>> resultTask, F
             return result.Ensure(predicate, errorMessage);
         }
 
-        public static async Task<Result<T>> Ensure<T, TError>(this Task<Result<T, TError>> resultTask,
+        public static async Task<Result<T, TError>> Ensure<T, TError>(this Task<Result<T, TError>> resultTask,
```

Comment:
I think prev returned result were invalid.
`Task<Result<T, TError>> -> Task<Result<T, TError>>` is correct flow